remote_rest/auth_utils.py

```diff
@@ -0,0 +1,103 @@
+import logging
+import os
+import secrets
+from datetime import datetime, timedelta, timezone
+from typing import Any, Dict, Optional
+
+from dotenv import load_dotenv
+from fastapi import Depends, HTTPException, status
+from fastapi.security import APIKeyHeader
+from passlib.context import CryptContext
+
+load_dotenv()
+
+
+TOKEN_EXPIRE_MINUTES = int(os.getenv('TOKEN_EXPIRE_MINUTES', 30))
+API_KEY_HEADER_NAME = 'X-Auth-Token'
+api_key_header_scheme = APIKeyHeader(name=API_KEY_HEADER_NAME, auto_error=True)
+pwd_context = CryptContext(schemes=['argon2'], deprecated='auto')
+
+# In-Memory token Store
+# token looks like this token : {"username": str, "expires": datetime}
+# state is lost on server restart.
+# Not inherently thread-safe for multi-worker setups without locks ?
```

Comment:
maybe we will switch at some point to jwt. I guess that would be thread-safe, since we don't have any state anymore